grakn-client/src/main/java/ai/grakn/remote/concept/RemoteAttributeType.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.remote.concept;
+
+import ai.grakn.concept.Attribute;
+import ai.grakn.concept.AttributeType;
+import ai.grakn.concept.ConceptId;
+import ai.grakn.remote.RemoteGraknTx;
+import com.google.auto.value.AutoValue;
+
+import javax.annotation.Nullable;
+
+/**
+ * @author Felix Chapman
+ *
+ * @param <D> The data type of this attribute type
+ */
+@AutoValue
```

Comment:
For this and all the other Remote concept maybe we want to override the equality to be based on the id? That's how the embedded API work and maybe this should do the same thing?

Reply:
Also maybe the current concepts can be autovalued? Not relevant to this PR of coourse

Reply:
I wondered about if we could autovalue the old concepts... maybe? It looks tough. `ConceptImpl` contains all these fields:

```
public abstract class ConceptImpl implements Concept, ConceptVertex, CacheOwner{
    private final Set<Cache> registeredCaches = new HashSet<>();
    //WARNING: DO not flush the current shard into the central cache. It is not safe to do so in a concurrent environment
    private final Cache<Shard> currentShard = Cache.createTxCache(this, Cacheable.shard(), () -> {
        String currentShardId = vertex().property(Schema.VertexProperty.CURRENT_SHARD);
        Vertex shardVertex = vertex().tx().getTinkerTraversal().V().has(Schema.VertexProperty.ID.name(), currentShardId).next();
        return vertex().tx().factory().buildShard(shardVertex);
    });
    private final Cache<Long> shardCount = Cache.createSessionCache(this, Cacheable.number(), () -> shards().count());
    private final Cache<ConceptId> conceptId = Cache.createPersistentCache(this, Cacheable.conceptId(), () -> ConceptId.of(vertex().property(Schema.VertexProperty.ID)));
    private final VertexElement vertexElement;
```

and they're not immutable and not part of object equality either... so it may be a bad idea.

Reply:
On equality, personally I think concepts from different transactions should not be equal - since they are not interchangeable with each other (imagine two concepts with the same ID from totally different keyspaces or even servers being equal).

On the other hand, the embedded concepts work this way and we _should_ make it consistent 🤷‍♂️

Reply:
Maybe we should be changing the embedded concepts to not use id? That sounds better based on your valid points?

Reply:
I think that's a good idea! Something _may_ subtly break - we'll see.

Reply:
actually I changed my mind everything is perfect as it is right now